dbaas/dbaas-aggregator/src/main/java/com/netcracker/cloud/dbaas/security/filters/KubernetesTokenAuthFilter.java

```diff
@@ -0,0 +1,23 @@
+package com.netcracker.cloud.dbaas.security.filters;
+
+import jakarta.ws.rs.client.ClientRequestContext;
+import jakarta.ws.rs.client.ClientRequestFilter;
+import jakarta.ws.rs.core.HttpHeaders;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+@Slf4j
+public class KubernetesTokenAuthFilter implements ClientRequestFilter {
+    private final Supplier<String> tokenSupplier;
+
+    public KubernetesTokenAuthFilter(Supplier<String> tokenSupplier) {
+        this.tokenSupplier = tokenSupplier;
+    }
+
+    @Override
+    public void filter(ClientRequestContext clientRequestContext) throws IOException {
+        clientRequestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, "Bearer " + tokenSupplier.get());
```

Comment:
Add null check for tokenSupplier.get() to prevent potential NullPointerException when token is null.

```suggestion
        String token = tokenSupplier.get();
        if (token != null) {
            clientRequestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, "Bearer " + token);
        }
```

#deepseek-review:inline